mmdet/datasets/custom.py

```diff
@@ -310,65 +311,81 @@ def evaluate(self,
             scale_ranges (list[tuple] | None): Scale ranges for evaluating mAP.
                 Default: None.
         """
-        # TODO[EUGENE]: ALLOW MULTIPLE METRICS
-        if not isinstance(metric, str):
-            assert len(metric) == 1
-            metric = metric[0]
-        allowed_metrics = ['mAP', 'recall', 'mIoU']
-        if metric not in allowed_metrics:
-            raise KeyError(f'metric {metric} is not supported')
-        annotations = [self.get_ann_info(i) for i in range(len(self))]
+        metrics = metric if isinstance(metric, list) else [metric]
+        allowed_metrics = ['mAP', 'recall', 'mIoU', 'mae', 'mae%']
         eval_results = OrderedDict()
-        iou_thrs = [iou_thr] if isinstance(iou_thr, float) else iou_thr
-        # TODO[EUGENE]: ADD MAE METRIC
-        if metric == 'mAP':
-            assert isinstance(iou_thrs, list)
-            mean_aps = []
-            for iou_thr in iou_thrs:
-                print_log(f'\n{"-" * 15}iou_thr: {iou_thr}{"-" * 15}')
-                if isinstance(results[0], tuple):
-                    mean_ap, _ = eval_segm(
+        for metric in metrics:
```

Comment:
There is a concern that CustomDataset is not custom. How about overriding evaluate function by inheriting an existing Dataset?

Reply:
We can move this to `MPADetDataset` or `OTEDataset`, but changing this will affect the work @goodsong81 is working on (replacement with openmmlab/mmdet).